cartographer_ros/src/msg_conversion.cc

```diff
@@ -151,40 +180,30 @@ sensor_msgs::LaserScan ToLaserScan(
 
 sensor_msgs::PointCloud2 ToPointCloud2Message(
     const int64 timestamp, const string& frame_id,
+    const ::cartographer::sensor::PointCloud& point_cloud) {
```

Comment:
We would first have to convert our point cloud to a PCL point cloud to do that. It doesn't look like PCL has anything equivalent to Eigen's Map, so that would require copying the data an additional time. Doesn't seem worth it.